website/blog/2022-10-18-how-to-design-metrics.md

```diff
@@ -0,0 +1,367 @@
+        - In a SaaS business though, it might make more sense - especially with usage based pricing. The Customer Success team might be interested in tracking the revenue of certain customers and ensuring that they remain consistent.
+
+To quote Cameron Afzal, Product Manager of the dbt Semantic Layer:
+
+> Thoughtful curation of dimensions is essential for three main reasons:
+- ****Relevance:**** Analysts must include the dimensions most relevant to answering the question.
+- ****Trust****: Curating high-quality dimensions with little to no known errors helps ensure trust in analysis results and the decisions that follow.
+- ****Efficiency****: Curation provides a faster path to high-quality analysis results.
+> 
+
+To put it another way, **metrics are most useful when every dimension provided can help provide answers to the business.** 
+
+### Advanced example - how to design complicated metrics
+
+Now let’s look at a more complicated example of a metric that is calculated from components that could be metrics unto themselves. The metric in question is Net Promoter score, which is used by the dbt Labs internal analytics team to understand the experience that users are having on dbt Cloud.
+
+For those of you who are unfamiliar with the industry metric of Net Promoter Score, here is a [great article from the folks over at Delighted on how it is calculated.](https://delighted.com/net-promoter-score) The short version of it is `the percentage of promoters - the percentage of detractors`. 
+
+---
+
+Here at dbt Labs we provide users with short surveys where they can provide feedback (as well as in a few other locations). The data is collected from those surveys is used to calculate our NPS Score, which helps us understand user sentiment over time.
+
+Given that these surveys come from a few different sources, there is a substantial amount of upstream modeling performed in order to unify them in a single model, but the end result is something that looks similar to the table below:
+
+| feedback_date | unique_id | feedback_source | user_type | account_plan | score | nps_category |
+| --- | --- | --- | --- | --- | --- | --- |
+| 2022-10-01 | 1 | nps_tool_1 | developer | team | 5 | detractor |
+| 2022-10-01 | 2 | nps_tool_2 | read_only | developer | 8 | promoter |
+| 2022-10-02 | 3 | nps_tool_1 | developer | enterprise | 10 | promoter |
+| 2022-10-02 | 4 | nps_tool_1 | developer | developer | 7 | passive |
+| 2022-10-02 | 5 | nps_tool_2 | developer | team | 9 | promoter |
+| 2022-10-03 | 6 | nps_tool_1 | developer | enterprise | 7 | passive |
+
+The dbt Internal Analytics team ([long may they reign](https://www.linkedin.com/feed/update/urn:li:activity:6962884130569080833/)) took this data and decided to build the NPS Score metric into our repo so that it could be surfaced to stakeholders in multiple tools. This process is where we began to form our opinions on what should live in the modeling layer vs semantic layer.
```

Comment:
I know I have said that we caveated in too many places, but I think that a caveat might be useful here along the lines of. "This process is where we began to form our early opinions on what should live in the modeling layer vs semantic layer - but these are sure to progress as we add in more and more real world use cases."